File: Terminus-Edition-2/aws-lambda-event-source-mapping-recovery/environment/src/paths.py
import os
from pathlib import Path

def app_root() -> Path:
    return Path(os.environ.get("APP_ROOT", "/app"))

def config_path(name: str) -> Path:
    return app_root() / "config" / name

def data_path(name: str) -> Path:
    return app_root() / "data" / name

def ledger_path() -> Path:
    return Path(os.environ.get("SIDE_EFFECT_LEDGER", str(data_path("side_effect_ledger.json"))))

def dlq_path() -> Path:
    return Path(os.environ.get("DLQ_STATE", str(data_path("dlq_state.json"))))

File: Terminus-Edition-2/aws-lambda-fix84-work/environment/simulator/paths.go
package main

import (
	"os"
	"path/filepath"
)

func appRoot() string {
	if value := os.Getenv("APP_ROOT"); value != "" {
		return value
	}
	return "/app"
}

func configPath(name string) string {
	return filepath.Join(appRoot(), "config", name)
}

func dataPath(name string) string {
	return filepath.Join(appRoot(), "data", name)
}

func ledgerPath() string {
	if value := os.Getenv("SIDE_EFFECT_LEDGER"); value != "" {
		return value
	}
	return dataPath("side_effect_ledger.json")
}

func dlqPath() string {
	if value := os.Getenv("DLQ_STATE"); value != "" {
		return value
	}
	return dataPath("dlq_state.json")
}
